util/ValueSetHelper.ts

```diff
@@ -0,0 +1,179 @@
+import { TestCaseInfo } from '../state/atoms/patientTestCase';
+import fhirpath from 'fhirpath';
+import { parsedCodePaths } from '../util/codePaths';
+import { DataRequirementsLookupByTypeProps } from '../state/selectors/dataRequirementsLookupByType';
+import { CodeableConcept } from 'fhir/r2';
+
+export interface GetValueSetCodesProps {
+  code?: string;
+  system?: string;
+}
+
+/**
+ * Helper function that takes a ValueSet url and a measure bundle and returns an
+ * array of objects that contain each code and system in the ValueSet
+ */
+function getValueSetCodes(valueSetUrl: string[], mb: fhir4.Bundle | null): GetValueSetCodesProps[] {
+  const codesAndSystems: GetValueSetCodesProps[] = [];
+  valueSetUrl.forEach(vs => {
+    const vsResource = mb?.entry?.filter(r => r.resource?.resourceType === 'ValueSet' && r.resource?.url === vs)[0]
+      .resource as fhir4.ValueSet;
+    vsResource.expansion?.contains?.forEach(c => {
+      codesAndSystems.push({ code: c.code, system: c.system });
+    });
+  });
+  return codesAndSystems;
+}
+
+/**
+ * Helper function for repeated logic in minimizeTestCaseResources that takes a primaryCodeValue,
+ * a resource, a matching DataRequirement of the same resourceType, a measureBundle, and a newResources
+ * array, checks that the primaryCodeValue exists and if the primaryCodeValue matches any of the
+ * directCodes or ValueSet codes of a DataRequirement of the same resourceType, the resource is added
+ * to the newResources array
+ */
+function checkCodesAndValueSets(
+  primaryCodeValue: fhir4.CodeableConcept,
+  matchingDRType: DataRequirementsLookupByTypeProps,
+  measureBundle: fhir4.Bundle | null,
+  resource: fhir4.BundleEntry,
+  newResources: fhir4.BundleEntry[]
+) {
+  if (primaryCodeValue) {
+    if (
+      matchingDRType.directCodes.length > 0 &&
+      matchingDRType.directCodes.find(dc => primaryCodeValue.coding?.find(c => c.code === dc.code))
```

Comment:
```suggestion
      matchingDRType.directCodes.find(dc => primaryCodeValue.coding?.includes(c => c.code === dc.code))
```

`includes` should be used since we only need to know about existence in the array.

Reply:
If I make this change, then I get a type error with the `.find` that it is in, should I change them both?

Reply:
Yep. Both would need to be `.includes`. and both predicates should be returning booleans.

Reply:
![image](https://github.com/user-attachments/assets/9fa99242-d109-4e5b-8d82-4347269a90a9)

Reply:
```suggestion
      matchingDRType.directCodes.some(dc => primaryCodeValue.coding?.some(c => c.code === dc.code))
```

[`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is actually what i was looking for this case.